iceoryx_dust/test/moduletests/test_cxx_string_conversion.cpp

```diff
@@ -108,4 +110,593 @@ TYPED_TEST(StdString_test, StringOfSizeCapaToSTDStringConvResultsInSizeCapa)
+    EXPECT_THAT(this->testSubject < testStdString, Eq(false));
+    EXPECT_THAT(this->testSubject <= testStdString, Eq(true));
+    EXPECT_THAT(this->testSubject > testStdString, Eq(false));
+    EXPECT_THAT(this->testSubject >= testStdString, Eq(true));
+
+    EXPECT_THAT(testStdString < this->testSubject, Eq(false));
+    EXPECT_THAT(testStdString <= this->testSubject, Eq(true));
+    EXPECT_THAT(testStdString > this->testSubject, Eq(false));
+    EXPECT_THAT(testStdString >= this->testSubject, Eq(true));
+}
+
+TYPED_TEST(StdString_test, CompareOperatorsWithDifferentStdStringWithDifferentSize)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "844fc675-25a2-415f-99c3-5eece0f2315d");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+
+    const std::string testStdString(STRINGCAP, 'L');
+
+    // compare with greater string
+    std::string temp1(STRINGCAP + 5U, 'M');
+    string<STRINGCAP + 5U> sutGreater;
+    ASSERT_THAT(sutGreater.unsafe_assign(temp1.c_str()), Eq(true));
+
+    EXPECT_THAT(sutGreater < testStdString, Eq(false));
+    EXPECT_THAT(sutGreater <= testStdString, Eq(false));
+    EXPECT_THAT(sutGreater > testStdString, Eq(true));
+    EXPECT_THAT(sutGreater >= testStdString, Eq(true));
+    EXPECT_THAT(testStdString < sutGreater, Eq(true));
+    EXPECT_THAT(testStdString <= sutGreater, Eq(true));
+    EXPECT_THAT(testStdString > sutGreater, Eq(false));
+    EXPECT_THAT(testStdString >= sutGreater, Eq(false));
+
+    // compare with less string
+    std::string temp2(STRINGCAP + 5U, 'F');
+    string<STRINGCAP + 5U> sutLess;
+    ASSERT_THAT(sutLess.unsafe_assign(temp2.c_str()), Eq(true));
+
+    EXPECT_THAT(sutLess < testStdString, Eq(true));
+    EXPECT_THAT(sutLess <= testStdString, Eq(true));
+    EXPECT_THAT(sutLess > testStdString, Eq(false));
+    EXPECT_THAT(sutLess >= testStdString, Eq(false));
+    EXPECT_THAT(testStdString < sutLess, Eq(false));
+    EXPECT_THAT(testStdString <= sutLess, Eq(false));
+    EXPECT_THAT(testStdString > sutLess, Eq(true));
+    EXPECT_THAT(testStdString >= sutLess, Eq(true));
+}
+
+TYPED_TEST(StdString_test, CompareOperatorsWithEqualStdStringWithDifferentCapa)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "e54181ac-6322-4b49-b26b-b17c7df1fe07");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+    std::string temp(STRINGCAP, 'M');
+    ASSERT_THAT(this->testSubject.unsafe_assign(temp.c_str()), Eq(true));
+
+    constexpr uint64_t TEST_CHAR_ARRAY_CAPACITY = STRINGCAP + 6U;
+    // NOLINTJUSTIFICATION required to verify string literal functionality of iox::string
+    // NOLINTNEXTLINE(hicpp-avoid-c-arrays, cppcoreguidelines-avoid-c-arrays)
+    char testCharArray[TEST_CHAR_ARRAY_CAPACITY];
+    for (auto& c : testCharArray)
+    {
+        c = 'M';
+    }
+    testCharArray[STRINGCAP] = '\0';
+
+    const std::string testStdString = &testCharArray[0];
```

Comment:
This doesn't make sense for a std::string since `testStdString`'s capacity doesn't have to be greater than `STRINGCAP` here. This could be achieved using `reserve` but I'm not sure if this test brings much value. Maybe we can remove it.